### PR TITLE
Writing Pseudocode for Project

### PR DESCRIPTION
**Pull request by martjor**

### CODE REVIEW

**Comment by martjor** [comment]
Hello @andrewfullard, 

I just wrote the blueprints for the scripts that my workflow uses as of today. The majority of these are separate python scripts, but I'll be treating them as functions for the purpose of writing the pseudocode.

I also copied the DFD into a Mermaid flowchart so that I can update it more easily as receive feedback from you.

**Comment by andrewfullard** [comment]
Looks like a good start! Looking forward to seeing the pseudocode fill out.

File: pseudocode [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    path_params
+        path to the .json file containing the miniaturization parameters for the graph
+    n_changes
+        number of changes proposed at each step
+    n_steps
+        number of total miniaturization steps
+    n_substeps
+        
+
+script simulate_sir
```
Comment (andrewfullard) [inline pseudocode:34]: Pseudocode is a little brief here in terms of inputs/outputs of each script

**Comment by andrewfullard** [comment]
Looking good, especially the snakefile. I would like to see some docstrings or other documentation of the snakefile/code.

**Pull request closed without merging**